src/jit/codegencommon.cpp

```diff
@@ -3981,10 +4001,12 @@ void            CodeGen::genFnPrologCalleeRegArgs(regNumber xtraReg,
         {
             // A struct might be passed  partially in XMM register for System V calls.
             // So a single arg might use both register files.
+#ifndef ARM_SOFTFP
             if (isFloatRegType(regType) != doingFloat)
```

Comment:
I'm not convinced that any of the changes to genFnPrologCalleeRegArgs(), or the new function TransformRegNumForArg(), are necessary. Because for softFP rsCalleeRegArgMaskLiveIn will be empty, genFnPrologCalleeRegArgs() won't be called for the FP registers. The incoming FP args will be in integer registers, and will probably get homed to some stack location. Later code that uses the arguments will do a floating-point register load from that stack location (this is perhaps not optimal, but that's a different story). Is that not already happening?